### ISSUE TITLE
Regression in Transform

### ISSUE THREAD

**Issue by Andykl** [renpy/renpy#3053]
This code behaves different in 7.4.(some) and 7.4.9
```python
label main_menu:
    return

init python:
    def execute_something():
        var = {"A": ["A", "a"]}
        # without try, this make exception even without renpy.transition
        # because screen execute its AST anyway by ScreenDisplayable._hide
        try:
            return var[current_screen]
        except:  
            return ["eval failed"]

screen A():
    tag common
    text "[current_screen] with transform":
        align (0.5, 0.5)
        at transform:
            zoom 1.0

    hbox:
        spacing 10
        align (0.5, 0.2)
        for t in execute_something():
            text "[t]"

screen B():
    tag common
    text "[current_screen] without transform":
        align (0.5, 0.5)

label start:
    $current_screen = "A"
    while True:
        $renpy.transition(Dissolve(0.5))
        if current_screen == "A":
            show screen A
        else:
            show screen B

        pause

        $current_screen = "A" if current_screen == "B" else "B"
    return
```
I triangled that to a3defaf. So, it happens only if a screen have a transform inside and the screen is replaced with different screen. So, removing transform from screen A makes it behave as intended. I'm not sure it actually a bug as this happens only if screen rely on some outside-of-screen context, which can be seen as a screen with side-effect. Passing the variables that screen uses explicitly solve the problem.

**Comment by Andykl** [comment]
If it's not a bug, I think it's a good idea to add a note to the screen documentation something like "It is a rule of thumb to pass to screens (including further passing to use screen if it is used) all non-constant variables that are used in the screen even if it's the same name".

**Comment by renpytom** [comment]
I believe this might be a regression.

**Comment by renpytom** [comment]
Fixed by f2868b02285d72bb36e8d7e498fc12b135b6753b (despite the bad commit message).
